fastcrypto/src/groups/bls12381.rs

```diff
@@ -135,6 +139,42 @@ impl GroupElement for G1Element {
     }
 }
 
+impl Pair for G1Element {
+    type Other = G2Element;
+    type Output = GTElement;
+
+    fn pair(&self, other: &Self::Other) -> Self::Output {
+        let mut self_affine = blst_p1_affine::default();
+        let mut other_affine = blst_p2_affine::default();
+        let mut res = blst_fp12::default();
+        unsafe {
+            blst_p1_to_affine(&mut self_affine, &self.0);
+            blst_p2_to_affine(&mut other_affine, &other.0);
+            blst_miller_loop(&mut res, &other_affine, &self_affine);
+            blst_final_exp(&mut res, &res);
+        }
+        Self::Output::from(res)
```

Comment:
here it defines (G1, G2) -> GT, are there ways to disallow definitions for "invalid" pairings? i.e. this trait should not implemented for certain type alias